test/domain-shared/git-utils.test.js

```diff
@@ -0,0 +1,65 @@
+describe("GitUtils", () => {
+    const scenarioModifiedOutput = {
+        modified: [
+            "package-lock.json",
+            "package.json",
+            "server_src/lib/serverCommands.js",
+        ],
+        ahead: 0,
+        behind: 0,
+    };
+
+    let wasReset = null;
+
+    jest.mock("simple-git", () => {
+        return () => {
+            return {
+                status: () => Promise.resolve(scenarioModifiedOutput),
+                pull: () => Promise.resolve(),
+                reset: (input) => {
+                    wasReset = true;
+                    return Promise.resolve();
+                },
+            };
+        };
+    });
+
+    const {
+        returnCurrentGitStatus,
+        isBranchUpToDate,
+        isBranchInfront,
+        getListOfModifiedFiles,
+        pullLatestRepository,
+    } = require("../../server_src/utils/git.utils");
```

Comment:
`require` after `jest.mock`. 
That's how jest is able to inject the mocked function, otherwise the mock would not be inserted and the real call would still happen. This is also a nice way to switch back and forth to a real implementation sometimes.